library/platform_util.c

```diff
@@ -211,29 +212,24 @@ static uint32_t mbedtls_platform_random( void )
                                &olen );
         if( olen == sizeof( hw_random ) )
         {
-            random_seed = hw_random % RAND_MODULUS;
+            random_state ^= hw_random;
```

Comment:
Why was this changed to a XOR?

Reply:
This change was suggested by @armask to do entropy mixing instead of changing the seed.